spacelift/resource_aws_role_test.go

```diff
@@ -75,16 +76,18 @@ func TestAWSRoleResource(t *testing.T) {
 				}
 
 				resource "spacelift_aws_role" "test" {
-					module_id = spacelift_module.test.id
-					role_arn  = "arn:aws:iam::039653571618:role/empty-test-role"
+					module_id        = spacelift_module.test.id
+					role_arn         = "arn:aws:iam::039653571618:role/empty-test-role"
+					duration_seconds = 942
 				}
 			`, randomID),
 				Check: Resource(
 					resourceName,
 					Attribute("id", IsNotEmpty()),
 					Attribute("module_id", Equals(fmt.Sprintf("test-module-%s", randomID))),
 					Attribute("generate_credentials_in_worker", Equals("false")),
-					AttributeNotPresent("external_id"),
+					Attribute("duration_seconds", Equals("942")),
+					Attribute("external_id", IsEmpty()),
```

Comment:
Why did you need to change this from `AttributeNotPresent()` to `IsEmpty()`?

Reply:
@adamconnelly I read the resource immediately after the creation of spacelift_aws_role, before we did not check what the state of unset values is - https://github.com/spacelift-io/terraform-provider-spacelift/pull/302/files#diff-21ecb1e1c4fcd6d7a2f835a4b4158104cf0d96cbe887c9b8960f51184688d9a7R125 . 

In the mutation, we set externalID to "" when externalID is missing.

Reply:
Cool - that makes sense.